src/ui-test/tests/remote.debug.on.openshift.test.ts

```diff
@@ -0,0 +1,245 @@
+
+            // open file
+            await VSBrowser.instance.openResources(path.join(QUARKUS_PROJECT_FOLDER, DEMO_FILE));
+
+            // set breakpoint
+            textEditor = new TextEditor();
+            await driver.wait(async function () {
+                return await textEditor.toggleBreakpoint(12);
+            }, 5000);
+
+            // open debug view
+            const debugView = await openDebugView();
+
+            // validate configuration and start debug
+            assert.isTrue(await validateLaunchConfiguration(debugView, QUARKUS_ATTACH_DEBUGGER));
+            await startDebugSession(debugView,QUARKUS_ATTACH_DEBUGGER);
+
+            const terminalView = await new BottomBarPanel().openTerminalView();
+
+            // check breakpoint is hitted
+            const breakpoint = await waitForBreakpointPause(driver, textEditor);
+            assert.isTrue(await breakpoint.isPaused());
+        });
+    });
+
+    /**
+     * Prepares the test environment by opening a directory in the VSCode instance and clearing its contents.
+     * 
+     * @param directory - The name of the directory to be opened in the Explorer view.
+     * @param directoryPath - The path of the directory to be cleared and opened in the Explorer view.
+     */
+    async function prepareEnvironment(directory: string, directoryPath: string): Promise<void> {
+        driver = VSBrowser.instance.driver
+        await VSBrowser.instance.openResources(directoryPath);
+        await deleteFolderContents(directoryPath);
+        await (await new ActivityBar().getViewControl('Explorer')).openView();
+        await new SideBarView().getContent().getSection(directory);
+        await workaround(driver);
+    }
+
+    /**
+     * Cleans up the environment by disconnecting the debugger, closing views, killing terminals, and clearing the directory.
+     * 
+     * @param directoryPath - The path of the directory to be cleared.
+     */
+    async function cleanUp(directoryPath: string): Promise<void> {
+        await disconnectDebugger(driver);
+        await (await new ActivityBar().getViewControl('Run and Debug')).closeView();
+        await killTerminal();
+        await new EditorView().closeAllEditors();
+        await deleteFolderContents(directoryPath);
+    }
+ });
+
+
+/**
+ * Provide workaround for Language Support for Java issue.
+ * https://github.com/redhat-developer/vscode-java/issues/3784
+ * 
+ * @param driver Active WebDriver isntance.
+ */
+async function workaround(driver: WebDriver): Promise<void> {
+    await executeCommand('Preferences: Open Workspace Settings (JSON)');
+    await waitUntilEditorIsOpened(driver, 'settings.json');
+    const editor = new TextEditor();
+    const newJson = addNewItemToRawJson(await editor.getText(), "java.project.sourcePaths", ["src/main/java"]);
+    await editor.setText(newJson);
+    await DefaultWait.sleep(10000);
+    await editor.save();
+    await DefaultWait.sleep(10000);
```

Comment:
waiting two times 10 seconds in 3 lines of code.
I think we should try to provide some waitUntil/driver.wait condition.

Have you already tried?
If yes, can you explain what have you tried and if you have ideas why it wasn't working?

Reply:
I was not able to solve failing pipeline in different way. There's problem only on CI, in local run there's no need to have any kind of wait. Issue with dynamic wait is that the condition is solved in no time so it acts like there's no timeout there. I've tried dynamic wait like this instead of first static timeout but it didn't help. Maybe there's a different way but I didn't found it. 
`await driver.wait(async function () {
    return (await editor.getText()).match(newJson);
}, 10000);`

Reply:
Maybe something like that can do the trick (tried only locally):
```
await driver.wait(async() => {
        return await editor.isDirty() && (await editor.getText()).replace(/\s+/g, '') === newJson.replace(/\s+/g, '');
    }, 10000);
await editor.save();
    await driver.wait(async() => {
        return !await editor.isDirty();
    }, 10000);
```

Reply:
Okay, this is working. Thank you very much for your investigation!